src/raftstore/store/worker/unsafe_cleanup_range.rs

```diff
@@ -0,0 +1,184 @@
+        .unwrap_or_else(|e| panic!("decode key failed: {:?}", e));
+    (cf, start_key)
+}
+
+impl TaskQueue {
+    pub fn new(db: Arc<DB>) -> TaskQueue {
+        TaskQueue { db: db }
+    }
+
+    pub fn add_task(&self, cf: &str, start_key: &[u8], end_key: &[u8]) -> Result<(), String> {
+        let task_key = encode_task_key(cf, start_key);
+        self.db.put(&task_key, end_key)?;
+        Ok(())
+    }
+
+    pub fn pick_task(&self) -> Option<Task> {
+        let mut read_options = ReadOptions::default();
+        read_options.fill_cache(false);
+        read_options.set_iterate_lower_bound(keys::UNSAFE_CLEANUP_RANGE_MIN_KEY);
+        read_options.set_iterate_upper_bound(keys::UNSAFE_CLEANUP_RANGE_MAX_KEY);
+        let mut iter = self.db.iter_opt(read_options);
+        iter.seek(SeekKey::Start);
+        if iter.valid() {
+            let (cf, start_key) = decode_task_key(iter.key());
+            Some(Task::new(
+                String::from(cf),
+                start_key,
+                iter.value().to_vec(),
+            ))
+        } else {
+            None
+        }
+    }
+
+    pub fn delete_task(&self, task: Task) -> Result<(), String> {
+        let task_key = encode_task_key(&task.cf, &task.start_key);
+        self.db.delete(&task_key)?;
+        Ok(())
+    }
+}
+
+pub struct Runner {
+    engine: Arc<DB>,
+    task_queue: Arc<TaskQueue>,
+    use_delete_range: bool,
+}
+
+impl Runner {
+    pub fn new(engine: Arc<DB>, task_queue: Arc<TaskQueue>, use_delete_range: bool) -> Runner {
+        Runner {
+            engine: engine,
+            task_queue: task_queue,
+            use_delete_range: use_delete_range,
+        }
+    }
+
+    fn unsafe_cleanup_range(&self, cf: &str, start_key: &[u8], end_key: &[u8]) {
+        let handle = get_cf_handle(&self.engine, cf).unwrap();
+
+        // Use delete_files_in_range to drop as many sst files as possible, this
+        // is a way to reclaim disk space quickly after drop a table/index.
+        self.engine
+            .delete_files_in_range_cf(handle, start_key, end_key, /* include_end */ false)
+            .unwrap_or_else(|e| {
+                panic!(
+                    "failed to delete files in range [{}, {}): {:?}",
+                    escape(start_key),
+                    escape(end_key),
+                    e
+                )
+            });
+
+        // Delete all remaining keys.
+        util::delete_all_in_range_cf(&self.engine, cf, start_key, end_key, self.use_delete_range)
+            .unwrap_or_else(|e| {
+                panic!(
+                    "failed to delete all in range [{}, {}), cf: {}, err: {:?}",
+                    escape(start_key),
+                    escape(end_key),
+                    cf,
+                    e
+                );
+            });
+    }
+
+    fn unsafe_cleanup_ranges(&self) {
+        let t = Instant::now();
+        while let Some(task) = self.task_queue.pick_task() {
```

Comment:
Maybe it's better to get all tasks by iterating, which means, call iterator's `next` method rather than create iterators and perform `seek` many times.